util/memory/action.go

```diff
@@ -43,16 +53,26 @@ func (a *LogOnExceed) Action(t *Tracker) {
 	defer a.mutex.Unlock()
 	if !a.acted {
 		a.acted = true
-		logutil.Logger(context.Background()).Warn("memory exceeds quota",
-			zap.Error(errMemExceedThreshold.GenWithStackByArgs(t.label, t.BytesConsumed(), t.bytesLimit, t.String())))
-		return
+		if a.logHook == nil {
+			logutil.Logger(context.Background()).Warn("memory exceeds quota",
+				zap.Error(errMemExceedThreshold.GenWithStackByArgs(t.label, t.BytesConsumed(), t.bytesLimit, t.String())))
+			return
+		}
+		a.logHook(a.ConnID)
```

Comment:
this log hook sends a message to a global channel whose capacity is 1024. If the oom threshold is too low and QPS is too high, a log of queries might be blocked on sending this message.